src/PayID/pay-id-client.ts

```diff
@@ -43,6 +52,140 @@ export default class PayIDClient {
    * @param useHttps Whether to use HTTPS when making PayID requests. Most users should set this to 'true' to avoid
    *                 Man-in-the-Middle attacks. Exposed as an option for testing purposes. Defaults to true.
    */
+  constructor(network: string, useHttps: boolean = true) {
+    this.wrappedPayIdClient = new PayIdClient(network, useHttps)
+  }
+
+  /**
+   * Retrieve the address associated with a PayID.
+   *
+   * @param payID The payID to resolve for an address.
+   * @returns An address representing the given PayID.
+   */
+  async addressForPayID(payID: string): Promise<CryptoAddressDetails> {
+    return this.wrappedPayIdClient.addressForPayId(payID)
+  }
+
+  /**
+   * Generate a new invoice with compliance requests.
+   *
+   * @param payID The Pay ID to request an invoice for.
+   * @param nonce A randomly selected nonce that is unique to this invoice request.
```

Comment:
For some reason, I thought nonce was no longer going to be used? I realize that's outside the scope of this review.

Reply:
Same comment here. Queried the PayID folks.

Reply:
Yeah, we're not using `nonce` anymore. I think the field we use now is `txId`, which gets provided by the PayID server on a `GET /payment-setup-details`, and which the client should provide for a `POST /payment-setup-details` and `POST /payment-proof`.